src/bin/bios.rs

```diff
@@ -86,6 +87,10 @@ fn bootloader_main(
     let max_phys_addr = e820_memory_map
         .iter()
         .map(|r| r.start_addr + r.len)
+        // Don't consider addresses > 4GiB when determining the maximum
+        // physical address for the bootloader, as we are in  protected mode and
+        // cannot address more than 4 GiB of memory anyway.
+        .filter(|&addr| addr < GIGABYTE * 4)
```

Comment:
It might be better to cap the address at 4 GiB instead of filtering the region out completely. For example, we don't want to filter a hypothetical region that goes from 2GiB to 5GiB.

For example: 

```suggestion
        .map(|&addr| u64::min(addr, GIGABYTE * 4))
```

Reply:
Or alternatively move the `u64::min` call around the whole expession calculating `max_phys_addr`.

Reply:
I wasn't sure whether we would prefer to potentially map only part of a region if it crosses the 4GiB boundary, or skip mapping that region entirely. I can change that!

Reply:
Done in 4da46e1a72b7279241eabf3f25abfb1f10fc69d7